src/main/scala/io/citrine/lolo/util/LoloPyDataLoader.scala

```diff
@@ -0,0 +1,126 @@
+package io.citrine.lolo.util
+
+import java.nio.ByteBuffer
+import java.nio.ByteOrder
+
+import io.citrine.lolo.PredictionResult
+
+/**
+  * Tool used to transfer data from LoloPy to the JVM
+  *
+  * Works by receiving an byte array and converting it to Scala Sequence
+  */
+object LoloPyDataLoader {
+
+  /**
+    * Receives the feature array. Assumes it to be in a C-ordering
+    * @param input Input array as a byte array
+    * @param numAttributes Number of features in the array
+    * @param bigEndian Whether the numbers are is big-endian or not
+    * @return The array as a Scala array
+    */
+  def getFeatureArray(input: Array[Byte], numAttributes : Integer, bigEndian: Boolean) : Seq[Vector[Double]] = {
+    // Get ordering
+    val ordering = if (bigEndian) ByteOrder.BIG_ENDIAN else ByteOrder.LITTLE_ENDIAN
+
+    // Wrap the input as a buffer
+    val buffer = ByteBuffer.wrap(input).order(ordering)
+
+    // Generate digits
+    val nDigits = input.length / 8
+    (0 until nDigits).map(x => buffer.getDouble(x * 8)).grouped(numAttributes).map(_.toVector).toVector
+  }
+
+
+  /**
+    * Receive a class or weight array
+    * @param input Input byte array
+    * @param getFloat Whether to read doubles (or integers)
+    * @param bigEndian Whether the file
+    */
+  def get1DArray(input: Array[Byte], getFloat: Boolean, bigEndian: Boolean) : Seq[Any] = {
+    // Get ordering
+    val ordering = if (bigEndian) ByteOrder.BIG_ENDIAN else ByteOrder.LITTLE_ENDIAN
+
+    // Make the buffer
+    val buffer = ByteBuffer.wrap(input).order(ordering)
+
+    // Convert to integer or float
+    if (getFloat) {
+      val nDigits = input.length / 8
+      (0 until nDigits).map(x => buffer.getDouble(x * 8)).toVector
+    } else {
+      val nDigits = input.length / 4
+      (0 until nDigits).map(x => buffer.getInt(x * 4)).toVector
+    }
+  }
+
+  /**
+    * Zips the features and labels together
+    * @param X Feature array
+    * @param y Label array
+    * @return Zipped arrays
+    */
+  def zipTrainingData(X: Seq[Vector[Double]], y: Seq[Any]) : Seq[(Vector[Double], Any)] = {
```

Comment:
Does this need to be stand-alone?

Reply:
Could you mention that's why you made it in the comment?

Reply:
Yep, that's a good idea. I've added an explanation